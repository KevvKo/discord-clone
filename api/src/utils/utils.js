require('dotenv').config();

const jwt = require('jsonwebtoken');
const { APP_SECRET } = process.env;

function getTokenPayload(token) {
  return jwt.verify(token, "deinemussa");
}

function getUserId(req, authToken) {
  if (req) {
    const authHeader = req.headers.authorization;
    if (authHeader) {
      const token = authHeader.replace('Bearer ', '');
      if (!token) {
        throw new Error('No token found');
      }
      const { userId } = getTokenPayload(token);
      return userId;
    }
  } else if (authToken) {
    const { userId } = getTokenPayload(authToken);
    return userId;
  }

  throw new Error('Not authenticated');
}

module.exports = {
  getUserId
};  