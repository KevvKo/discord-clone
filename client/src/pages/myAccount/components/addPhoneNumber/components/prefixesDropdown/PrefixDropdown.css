.prefix-dropdown{
    --dropdown-menu-width: 210px;
}

.prefix-dropdown .dropdown-menu{
    width: var(--dropdown-menu-width) !important;
}