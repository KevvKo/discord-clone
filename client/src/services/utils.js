// Array 

/**
 * @description returns a sequence as array
 * @param {integer} start 
 * @param {Integer} stop 
 * @param {Integer} step 
 */

export const range = (start, stop, step = 1) => {
        
    return Array.from(
        { length: (stop - start) / step + 1}, 
        (_, i) => start + (i * step)
    );
};