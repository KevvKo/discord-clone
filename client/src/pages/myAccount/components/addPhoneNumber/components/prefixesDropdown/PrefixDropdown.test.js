import React from 'react';
import { render, screen, fireEvent, act } from '../../../../../../scripts/utils';
import PrefixDropdown from './PrefixDropdown';
import renderer from 'react-test-renderer';

describe('PrefixDropdown component', () => {
    let component;
    const mockCallback = jest.fn();

    beforeEach(() => {
        component = renderer.create( <PrefixDropdown onSelect={mockCallback}/> );
    });

    it('should render correctly', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
});
