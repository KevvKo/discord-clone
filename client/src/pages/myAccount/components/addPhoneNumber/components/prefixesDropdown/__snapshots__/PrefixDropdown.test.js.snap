// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrefixDropdown component should render correctly 1`] = `
<div
  className="prefix-dropdown dropdown"
  name="test"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    +31
  </button>
</div>
`;
